### DISCUSSION TITLE
keep-alive app 隐藏时是否可以不进行 release effect 行为？

### DISCUSSION THREAD

**Discussion by refinist** [jd-opensource/micro-app#1299] [General]
0.x 时只针对应用卸载时处理 release effect，当 keep-alive app hidden 时，推入后台但是仍然会运行脚本（一些通信监听代码），以往我们在 0.x 时就依赖了这一特性。但 1.x 在 keep-alive hidden 时处理了 release effect 导致后台不能运行
是否可以有个开关类似 hidden-keep-alive-app-release 控制 keep-alive hidden 时是否进行 release effect，在代码 https://github.com/micro-zoe/micro-app/blob/eaf044da70b0f7e5ffac6b3acd0b8252b6ac61a3/src/create_app.ts#L626-L639 
修改 638 行
```js
if ([undefined, true].includes(microApp.options['hidden-keep-alive-app-release'])) { 
  this.sandBox?.recordAndReleaseEffect({ keepAlive: true }) 
}
```
提问：keep-alive app hidden 禁止 release effect 会导致什么问题吗？如果开发者对于自己的程序足够了解是否可以有个这个开关呢？

**Comment by bailicangdu** [comment]
隐藏时不做任何限制会导致很多问题，这种场景比较少，后续可能会增加相应的配置项，只隐藏不限制行为
